plugins/out_opentelemetry/opentelemetry.c

```diff
@@ -1080,6 +1080,71 @@ static int append_v1_logs_metadata(struct opentelemetry_context *ctx,
     return 0;
 }
 
+}
+
+static int append_v1_logs_message(struct opentelemetry_context *ctx,
+                                   struct flb_log_event *event,
+                                   Opentelemetry__Proto__Logs__V1__LogRecord  *log_record)
+{
+    struct flb_ra_value *ra_val;
+
+    flb_plg_info(ctx->ins, "Appending logs for message");
+
+    if (ctx == NULL || event == NULL || log_record == NULL) {
+        return -1;
+    }
+
+    /* SpanId */
+    if (ctx->ra_span_id_message) {
+        flb_plg_info(ctx->ins, "pattern is %s\n", ctx->ra_span_id_message->pattern);
+        ra_val = flb_ra_get_value_object(ctx->ra_span_id_message, *event->body);
+        flb_plg_info(ctx->ins, "ra_val type is %d\n", (int)ra_val->o.type);
+        if (ra_val != NULL) {
+            if(ra_val->o.type == MSGPACK_OBJECT_BIN){
+                flb_plg_info(ctx->ins, "span id ra_val bin");
+                log_record->span_id.data = flb_calloc(1, ra_val->o.via.bin.size);
+                if (log_record->span_id.data) {
+                    flb_plg_info(ctx->ins, "span id has data");
+                    memcpy(log_record->span_id.data, ra_val->o.via.bin.ptr, ra_val->o.via.bin.size);
+                    log_record->span_id.len = ra_val->o.via.bin.size;
+                }
+            }else if(ra_val->o.type == MSGPACK_OBJECT_STR){
```

Comment:
Front end / collectors reject the request with “unmarshal: invalid SpanID length”.